nixos/modules/services/monitoring/cockpit.nix

```diff
@@ -0,0 +1,225 @@
+    systemd.services."cockpit-wsinstance-https@" = {
+      description = "Cockpit Web Service https instance %I";
+      bindsTo = [ "cockpit.service"];
+      path = [ cfg.package ];
+      documentation = [ "man:cockpit-ws(8)" ];
+      serviceConfig = {
+        Slice = "system-cockpithttps.slice";
+        ExecStart = "${cfg.package}/libexec/cockpit-ws --for-tls-proxy --port=0";
+        User = "root";
+        Group = "";
+      };
+    };
+
+    # Translation from $out/lib/systemd/system/cockpit-wsinstance-http.socket
+    systemd.sockets.cockpit-wsinstance-http = {
+      unitConfig = {
+        Description="Socket for Cockpit Web Service http instance";
+        BindsTo="cockpit.service";
+        Documentation="man:cockpit-ws(8)";
+      };
+      socketConfig = {
+        ListenStream="/run/cockpit/wsinstance/http.sock";
+        SocketUser="root";
+        SocketMode="0600";
+      };
+    };
+
+    # Translation from $out/lib/systemd/system/cockpit-wsinstance-https-factory.socket
+    systemd.sockets.cockpit-wsinstance-https-factory = {
+      unitConfig = {
+        Description="Socket for Cockpit Web Service https instance factory";
+        BindsTo="cockpit.service";
+        Documentation="man:cockpit-ws(8)";
+      };
+      socketConfig = {
+        ListenStream="/run/cockpit/wsinstance/https-factory.sock";
+        Accept=true;
+        SocketUser="root";
+        SocketMode="0600";
+      };
+    };
+
+    # Translation from $out/lib/systemd/system/cockpit-wsinstance-https-factory@.service
+    systemd.services."cockpit-wsinstance-https-factory@" = {
+      description = "Cockpit Web Service https instance factory";
+      documentation = [ "man:cockpit-ws(8)" ];
+      path = [ cfg.package ];
+      serviceConfig = {
+        ExecStart = "${cfg.package}/libexec/cockpit-wsinstance-factory";
+        User = "root";
+      };
+    };
+
+    # Translation from $out/lib/systemd/system/cockpit-wsinstance-http.service
+    systemd.services."cockpit-wsinstance-http" = {
+      description = "Cockpit Web Service http instance";
+      bindsTo = [ "cockpit.service" ];
+      path = [ cfg.package ];
+      documentation = [ "man:cockpit-ws(8)" ];
+      serviceConfig = {
+        ExecStart = "${cfg.package}/libexec/cockpit-ws --no-tls --port=0";
+        User = "root";
+        Group = "";
+      };
+    };
+
+    # Translation from $out/lib/systemd/system/cockpit.socket
+    systemd.sockets."cockpit" = {
+      unitConfig = {
+        Description="Cockpit Web Service Socket";
+        Documentation="man:cockpit-ws(8)";
+        Wants="cockpit-motd.service";
+      };
+      socketConfig = {
+        ListenStream=cfg.port;
+        ExecStartPost= [
+          "-${cfg.package}/share/cockpit/motd/update-motd \"\" localhost"
+          "-${pkgs.coreutils}/bin/ln -snf active.motd /run/cockpit/motd"
+        ];
+        ExecStopPost="-${pkgs.coreutils}/bin/ln -snf inactive.motd /run/cockpit/motd";
```

Comment:
If the program should not modify this we should use chattr if possible.

Reply:
It only changes /run/cockpit/active.motd

active.motd is a file
inactive.motd is a symlink to a immutable file in the nix store